src/cressi_goa.c

```diff
@@ -506,5 +620,35 @@ cressi_goa_device_foreach (dc_device_t *abstract, dc_dive_callback_t callback, v
 error_free_logbook:
 	dc_buffer_free(logbook);
 error_exit:
+	dc_buffer_free(conf.id);
+	return status;
+}
+
+static dc_status_t cressi_goa_device_timesync (dc_device_t *abstract, const dc_datetime_t *datetime)
+{
+	dc_status_t status = DC_STATUS_SUCCESS;
+	cressi_goa_device_t *device = (cressi_goa_device_t *) abstract;
+
+	unsigned char new_time[7];
+	array_uint16_le_set(new_time, datetime->year);
+	new_time[2] = datetime->month;
+	new_time[3] = datetime->day;
+	new_time[4] = datetime->hour;
+	new_time[5] = datetime->minute;
+	new_time[6] = datetime->second;
+	status = cressi_goa_device_transfer (device, CMD_SET_TIME, new_time, 7, NULL, NULL, NULL);
+	if (status != DC_STATUS_SUCCESS) {
+		ERROR (abstract->context, "Failed to set the new time.");
```

Comment:
Add a `return status;` statement here. I know it doesn't really make a difference now, but it helps preventing mistakes if this code is changed in the future.